crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -1178,6 +1178,17 @@ impl<'w> EntityWorldMut<'w> {
         self
     }
 
+    /// Calls [`EntityWorldMut::remove_by_id`] on every component in the entity.
+    ///
+    /// Allows you to clean-up all components associated with the entity without removing it from the world.
+    pub fn clear(&mut self) {
+        // We must collect here to appease the borrow checker
+        let components: Vec<ComponentId> = self.archetype().components().collect();
+        for component_id in components {
+            self.remove_by_id(component_id);
```

Comment:
I feel like we should drop down a level and use `remove_bundle` here to skip the repeated checks that the component exists.

Reply:
Should be addressed